src/model.ts

```diff
@@ -15,7 +15,53 @@
  */
 
 import * as common from '@google-cloud/common';
-import {Dataset} from './dataset';
+import {promisifyAll} from '@google-cloud/promisify';
+import arrify = require('arrify');
+import * as extend from 'extend';
+import {
+  BigQuery,
+  Job,
+  Dataset,
+  ResourceCallback,
+  RequestCallback,
+  JobRequest,
+} from '.';
+import {JobMetadata} from './job';
+import bigquery from './types';
+
+// This is supposed to be a @google-cloud/storage `File` type. The storage npm
```

Comment:
Great comment, thank you for being this explicit.  I kinda think about this problem a lot.  Could I trouble you to file a top level tracking issue in https://github.com/googleapis/google-cloud-node on this problem?